crates/bitwarden-ssh/src/import.rs

```diff
@@ -0,0 +1,279 @@
+use ed25519;
+use pkcs8::{
+    der::Decode, EncryptedPrivateKeyInfo, ObjectIdentifier, PrivateKeyInfo, SecretDocument,
+};
+use serde::{Deserialize, Serialize};
+use ssh_key::{
+    private::{Ed25519Keypair, RsaKeypair},
+    HashAlg, LineEnding,
+};
+#[cfg(feature = "wasm")]
+use tsify_next::Tsify;
+
+use crate::{error::SshKeyImportError, SshKey};
+
+const PKCS1_HEADER: &str = "-----BEGIN RSA PRIVATE KEY-----";
+const PKCS8_UNENCRYPTED_HEADER: &str = "-----BEGIN PRIVATE KEY-----";
+const PKCS8_ENCRYPTED_HEADER: &str = "-----BEGIN ENCRYPTED PRIVATE KEY-----";
+const OPENSSH_HEADER: &str = "-----BEGIN OPENSSH PRIVATE KEY-----";
+
+pub const RSA_PKCS8_ALGORITHM_OID: ObjectIdentifier =
+    ObjectIdentifier::new_unwrap("1.2.840.113549.1.1.1");
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+#[cfg_attr(feature = "wasm", derive(Tsify), tsify(into_wasm_abi, from_wasm_abi))]
+enum KeyType {
+    Ed25519,
+    Rsa,
+    Unknown,
+}
+
+pub fn import_key(encoded_key: String, password: String) -> Result<SshKey, SshKeyImportError> {
```

Comment:
**question:** Should password be an `Option`? Relying on empty string is a bit less conventional.